packages/horizon/contracts/data-service/extensions/DataServiceFees.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.26;
+
+import { IDataServiceFees } from "../interfaces/IDataServiceFees.sol";
+
+import { ProvisionTracker } from "../libraries/ProvisionTracker.sol";
+import { LinkedList } from "../../libraries/LinkedList.sol";
+
+import { DataService } from "../DataService.sol";
+import { DataServiceFeesV1Storage } from "./DataServiceFeesStorage.sol";
+
+/**
+ * @title DataServiceFees contract
+ * @dev Implementation of the {IDataServiceFees} interface.
+ * @notice Extension for the {IDataService} contract to handle payment collateralization
+ * using a Horizon provision. See {IDataServiceFees} for more details.
+ */
+abstract contract DataServiceFees is DataService, DataServiceFeesV1Storage, IDataServiceFees {
+    using ProvisionTracker for mapping(address => uint256);
+    using LinkedList for LinkedList.List;
+
+    /**
+     * @notice See {IDataServiceFees-releaseStake}
+     */
+    function releaseStake(uint256 n) external virtual {
```

Comment:
shouldn't this have `override`?

Reply:
yep good catch. since it's an abstract contract the lack of override wasn't showing as an error